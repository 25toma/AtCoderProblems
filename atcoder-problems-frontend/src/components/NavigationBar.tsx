import * as React from "react";
import { Navbar, Nav, NavItem, NavDropdown, MenuItem } from "react-bootstrap";
import { Arguments } from "../utils/Arguments";

interface NavigationBarProps {
  args: Arguments;
}
/**
 * The navigation bar which is always shown.
 * It doesn't have any status or props.
 */
export class NavigationBar extends React.Component<NavigationBarProps, {}> {
  render() {
    return (
      <Navbar inverse collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <a href={`./?user=${this.props.args.userId}`}>AtCoder Problems</a>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <NavDropdown title="Rankings" id="basic-nav-dropdown">
              <MenuItem href="./ac">Accepted Count</MenuItem>
              <MenuItem href="./fast">Fastest Codes</MenuItem>
              <MenuItem href="./short">Shortest Codes</MenuItem>
              <MenuItem href="./first">First Acceptances</MenuItem>
            </NavDropdown>
            <MenuItem href={`./?user=${this.props.args.userId}&kind=user`}>
              User Page
            </MenuItem>
            <NavDropdown title="Links" id="basic-nav-dropdown">
              <MenuItem href="https://atcoder.jp/">
                AtCoder Official Site
              </MenuItem>
              <MenuItem href="http://ichyo.jp/aoj-icpc/">AOJ-ICPC</MenuItem>
              <MenuItem href="https://github.com/kenkoooo/AtCoderProblems">
                Source Code (GitHub)
              </MenuItem>
              <MenuItem href="https://twitter.com/kenkoooo">@kenkoooo</MenuItem>
            </NavDropdown>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}
