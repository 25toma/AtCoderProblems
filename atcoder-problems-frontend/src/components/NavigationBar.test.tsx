import * as React from "react";
import { shallow, mount } from "enzyme";
import { configure } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import { NavigationBar } from "./NavigationBar";

configure({ adapter: new Adapter() });

test("render navigation bar", () => {
  let wrapper = mount(
    <NavigationBar
      args={{
        userId: "kenkoooo",
        rivals: ["chokudai", "iwiwi"],
        kind: "category",
        ranking: ""
      }}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
