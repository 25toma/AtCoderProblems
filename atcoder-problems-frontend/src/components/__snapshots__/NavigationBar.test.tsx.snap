// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navigation bar 1`] = `
<NavigationBar
  args={
    Object {
      "kind": "category",
      "ranking": "",
      "rivals": Array [
        "chokudai",
        "iwiwi",
      ],
      "userId": "kenkoooo",
    }
  }
>
  <Uncontrolled(Navbar)
    bsStyle="default"
    collapseOnSelect={true}
    inverse={true}
  >
    <Navbar
      bsClass="navbar"
      bsStyle="default"
      collapseOnSelect={true}
      componentClass="nav"
      fixedBottom={false}
      fixedTop={false}
      fluid={false}
      inverse={true}
      onToggle={[Function]}
      staticTop={false}
    >
      <nav
        className="navbar navbar-inverse"
      >
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <div
            className="container"
          >
            <NavbarHeader>
              <div
                className="navbar-header"
              >
                <NavbarBrand>
                  <a
                    className="navbar-brand"
                    href="./?user=kenkoooo"
                  >
                    AtCoder Problems
                  </a>
                </NavbarBrand>
                <NavbarToggle>
                  <button
                    className="navbar-toggle collapsed"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      className="sr-only"
                    >
                      Toggle navigation
                    </span>
                    <span
                      className="icon-bar"
                    />
                    <span
                      className="icon-bar"
                    />
                    <span
                      className="icon-bar"
                    />
                  </button>
                </NavbarToggle>
              </div>
            </NavbarHeader>
            <NavbarCollapse>
              <Collapse
                dimension="height"
                getDimensionValue={[Function]}
                in={false}
                mountOnEnter={false}
                timeout={300}
                transitionAppear={false}
                unmountOnExit={false}
              >
                <Transition
                  aria-expanded={null}
                  className=""
                  enteredClassName="collapse in"
                  enteringClassName="collapsing"
                  exitedClassName="collapse"
                  exitingClassName="collapsing"
                  in={false}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  transitionAppear={false}
                  unmountOnExit={false}
                >
                  <div
                    aria-expanded={null}
                    className="navbar-collapse collapse"
                  >
                    <Nav
                      bsClass="nav"
                      justified={false}
                      pullLeft={false}
                      pullRight={false}
                      stacked={false}
                    >
                      <ul
                        className="nav navbar-nav"
                        role={null}
                      >
                        <NavDropdown
                          id="basic-nav-dropdown"
                          key=".0"
                          onSelect={[Function]}
                          title="Rankings"
                        >
                          <Uncontrolled(Dropdown)
                            className=""
                            componentClass="li"
                            id="basic-nav-dropdown"
                            onSelect={[Function]}
                          >
                            <Dropdown
                              bsClass="dropdown"
                              className=""
                              componentClass="li"
                              id="basic-nav-dropdown"
                              onSelect={[Function]}
                              onToggle={[Function]}
                            >
                              <li
                                className="dropdown"
                              >
                                <DropdownToggle
                                  bsClass="dropdown-toggle"
                                  bsRole="toggle"
                                  id="basic-nav-dropdown"
                                  key=".0"
                                  onClick={[Function]}
                                  onKeyDown={[Function]}
                                  open={false}
                                  useAnchor={true}
                                >
                                  <SafeAnchor
                                    aria-expanded={false}
                                    aria-haspopup={true}
                                    className="dropdown-toggle"
                                    componentClass="a"
                                    id="basic-nav-dropdown"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="button"
                                  >
                                    <a
                                      aria-expanded={false}
                                      aria-haspopup={true}
                                      className="dropdown-toggle"
                                      href="#"
                                      id="basic-nav-dropdown"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="button"
                                    >
                                      Rankings
                                       
                                      <span
                                        className="caret"
                                      />
                                    </a>
                                  </SafeAnchor>
                                </DropdownToggle>
                                <DropdownMenu
                                  bsClass="dropdown-menu"
                                  bsRole="menu"
                                  key=".1"
                                  labelledBy="basic-nav-dropdown"
                                  onClose={[Function]}
                                  onSelect={[Function]}
                                  pullRight={false}
                                >
                                  <RootCloseWrapper
                                    disabled={true}
                                    event="click"
                                    onRootClose={[Function]}
                                  >
                                    <ul
                                      aria-labelledby="basic-nav-dropdown"
                                      className="dropdown-menu"
                                      role="menu"
                                    >
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="./?kind=ranking&ranking=ac"
                                        key=".$.0"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="./?kind=ranking&ranking=ac"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="./?kind=ranking&ranking=ac"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              Accepted Count
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="./?kind=ranking&ranking=fast"
                                        key=".$.1"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="./?kind=ranking&ranking=fast"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="./?kind=ranking&ranking=fast"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              Fastest Codes
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="./?kind=ranking&ranking=short"
                                        key=".$.2"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="./?kind=ranking&ranking=short"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="./?kind=ranking&ranking=short"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              Shortest Codes
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="./?kind=ranking&ranking=first"
                                        key=".$.3"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="./?kind=ranking&ranking=first"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="./?kind=ranking&ranking=first"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              First Acceptances
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="./?kind=ranking&ranking=sums"
                                        key=".$.4"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="./?kind=ranking&ranking=sums"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="./?kind=ranking&ranking=sums"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              Rated-Point
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                    </ul>
                                  </RootCloseWrapper>
                                </DropdownMenu>
                              </li>
                            </Dropdown>
                          </Uncontrolled(Dropdown)>
                        </NavDropdown>
                        <MenuItem
                          bsClass="dropdown"
                          disabled={false}
                          divider={false}
                          header={false}
                          href="./?user=kenkoooo&kind=user"
                          key=".1"
                          onSelect={[Function]}
                        >
                          <li
                            className=""
                            role="presentation"
                          >
                            <SafeAnchor
                              componentClass="a"
                              href="./?user=kenkoooo&kind=user"
                              onClick={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                            >
                              <a
                                href="./?user=kenkoooo&kind=user"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                role="menuitem"
                                tabIndex="-1"
                              >
                                User Page
                              </a>
                            </SafeAnchor>
                          </li>
                        </MenuItem>
                        <NavDropdown
                          id="basic-nav-dropdown"
                          key=".2"
                          onSelect={[Function]}
                          title="Links"
                        >
                          <Uncontrolled(Dropdown)
                            className=""
                            componentClass="li"
                            id="basic-nav-dropdown"
                            onSelect={[Function]}
                          >
                            <Dropdown
                              bsClass="dropdown"
                              className=""
                              componentClass="li"
                              id="basic-nav-dropdown"
                              onSelect={[Function]}
                              onToggle={[Function]}
                            >
                              <li
                                className="dropdown"
                              >
                                <DropdownToggle
                                  bsClass="dropdown-toggle"
                                  bsRole="toggle"
                                  id="basic-nav-dropdown"
                                  key=".0"
                                  onClick={[Function]}
                                  onKeyDown={[Function]}
                                  open={false}
                                  useAnchor={true}
                                >
                                  <SafeAnchor
                                    aria-expanded={false}
                                    aria-haspopup={true}
                                    className="dropdown-toggle"
                                    componentClass="a"
                                    id="basic-nav-dropdown"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="button"
                                  >
                                    <a
                                      aria-expanded={false}
                                      aria-haspopup={true}
                                      className="dropdown-toggle"
                                      href="#"
                                      id="basic-nav-dropdown"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="button"
                                    >
                                      Links
                                       
                                      <span
                                        className="caret"
                                      />
                                    </a>
                                  </SafeAnchor>
                                </DropdownToggle>
                                <DropdownMenu
                                  bsClass="dropdown-menu"
                                  bsRole="menu"
                                  key=".1"
                                  labelledBy="basic-nav-dropdown"
                                  onClose={[Function]}
                                  onSelect={[Function]}
                                  pullRight={false}
                                >
                                  <RootCloseWrapper
                                    disabled={true}
                                    event="click"
                                    onRootClose={[Function]}
                                  >
                                    <ul
                                      aria-labelledby="basic-nav-dropdown"
                                      className="dropdown-menu"
                                      role="menu"
                                    >
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="https://atcoder.jp/"
                                        key=".$.0"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="https://atcoder.jp/"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="https://atcoder.jp/"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              AtCoder Official Site
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="http://ichyo.jp/aoj-icpc/"
                                        key=".$.1"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="http://ichyo.jp/aoj-icpc/"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="http://ichyo.jp/aoj-icpc/"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              AOJ-ICPC
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="https://github.com/kenkoooo/AtCoderProblems"
                                        key=".$.2"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="https://github.com/kenkoooo/AtCoderProblems"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="https://github.com/kenkoooo/AtCoderProblems"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              Source Code (GitHub)
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                      <MenuItem
                                        bsClass="dropdown"
                                        disabled={false}
                                        divider={false}
                                        header={false}
                                        href="https://twitter.com/kenkoooo"
                                        key=".$.3"
                                        onKeyDown={[Function]}
                                        onSelect={[Function]}
                                      >
                                        <li
                                          className=""
                                          role="presentation"
                                        >
                                          <SafeAnchor
                                            componentClass="a"
                                            href="https://twitter.com/kenkoooo"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="menuitem"
                                            tabIndex="-1"
                                          >
                                            <a
                                              href="https://twitter.com/kenkoooo"
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              role="menuitem"
                                              tabIndex="-1"
                                            >
                                              @kenkoooo
                                            </a>
                                          </SafeAnchor>
                                        </li>
                                      </MenuItem>
                                    </ul>
                                  </RootCloseWrapper>
                                </DropdownMenu>
                              </li>
                            </Dropdown>
                          </Uncontrolled(Dropdown)>
                        </NavDropdown>
                      </ul>
                    </Nav>
                  </div>
                </Transition>
              </Collapse>
            </NavbarCollapse>
          </div>
        </Grid>
      </nav>
    </Navbar>
  </Uncontrolled(Navbar)>
</NavigationBar>
`;
